cantabular/dimensions.go

```diff
@@ -224,3 +227,67 @@ func (c *Client) GetParents(ctx context.Context, req GetParentsRequest) (*GetPar
 
 	return &resp.Data, nil
 }
+
+// GetGeographyDimensionsInBatches performs a graphQL query to obtain all the geography dimensions for the provided cantabular dataset.
+// The whole response is loaded to memory.
+func (c *Client) GetGeographyDimensionsInBatches(ctx context.Context, datasetID string, batchSize, maxWorkers int) (dataset *gql.Dataset, err error) {
+
+	// reference GetInstanceDimensionsInBatches
+	var processBatch GetGeographyBatchProcessor = func(b *GetGeographyDimensionsResponse) (abort bool, err error) {
+		if dataset == nil {
+			dataset = &gql.Dataset{}
+			dataset.RuleBase.Name = b.Dataset.RuleBase.Name
+			dataset.RuleBase.IsSourceOf.Search = b.Dataset.RuleBase.IsSourceOf.Search
+			dataset.RuleBase.IsSourceOf.CategorySearch = b.Dataset.RuleBase.IsSourceOf.CategorySearch
+			dataset.RuleBase.IsSourceOf.TotalCount = b.Dataset.RuleBase.IsSourceOf.TotalCount
+
+			dataset.RuleBase.IsSourceOf.Edges = make([]gql.Edge, dataset.RuleBase.IsSourceOf.TotalCount)
+		}
+
+		for i := 0; i < len(b.Dataset.RuleBase.IsSourceOf.Edges); i++ {
+			dataset.RuleBase.IsSourceOf.Edges[i+b.PaginationResponse.Offset] = b.Dataset.RuleBase.IsSourceOf.Edges[i]
+		}
+		return false, nil
+	}
+
+	// call GetGeographyBatchProcess in batches and aggregate the responses
+	err = c.GetGeographyBatchProcess(ctx, datasetID, processBatch, batchSize, maxWorkers)
+	if err != nil {
+		return nil, err
```

Comment:
Please wrap errors with context using errors.Wrap. Also this could be done in one line `if err := c.Get....; err != nil{...`